x/feegrant/keeper/msg_server.go

```diff
@@ -23,6 +23,27 @@ func NewMsgServerImpl(k Keeper) types.MsgServer {
 
 var _ types.MsgServer = msgServer{}
 
+func (k msgServer) validAllowedMsgs(msg *types.MsgGrantFeeAllowance) error {
+	f, err := msg.GetFeeAllowanceI()
+	if err != nil {
+		return err
+	}
+
+	allowedFeeAllowance, ok := f.(*types.AllowedMsgFeeAllowance)
+	if ok {
+		msgs := allowedFeeAllowance.AllowedMessages
+
+		for _, m := range msgs {
+			// If the granted service Msg doesn't exist, we throw an error.
+			if k.msgSvcRouter.Handler(m) == nil {
+				return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "%s doesn't exist", m)
+			}
+		}
+	}
+
```

Comment:
This all seems unnecessary. IMHO we don't need to do this type of validation. It just makes things more complicated and doesn't add a lot of value.
```suggestion
```